txscript/pkscript_test.go

```diff
@@ -0,0 +1,390 @@
+			class:   ScriptHashTy,
+			pkScript: []byte{
+				// OP_HASH160
+				0xa9,
+				// OP_DATA_20
+				0x14,
+				// <20-byte script hash>
+				0x90, 0x1c, 0x86, 0x94, 0xc0, 0x3f, 0xaf, 0xd5,
+				0x52, 0x28, 0x10, 0xe0, 0x33, 0x0f, 0x26, 0xe6,
+				0x7a, 0x85, 0x33, 0xcd,
+				// OP_EQUAL
+				0x87,
+			},
+		},
+		// Valid P2SH.
+		{
+			sigScript: []byte{0x6b, 0x65, 0x6b}, // kek
+			witness:   nil,
+			class:     ScriptHashTy,
+			pkScript: []byte{
+				// OP_HASH160
+				0xA9,
+				// OP_DATA_20
+				0x14,
+				// <20-byte script hash>
+				0xec, 0x6f, 0x7a, 0x5a, 0xa8, 0xf2, 0xb1, 0x0c,
+				0xa5, 0x15, 0x04, 0x52, 0x3a, 0x60, 0xd4, 0x03,
+				0x06, 0xf6, 0x96, 0xcd,
+				// OP_EQUAL
+				0x87,
+			},
+		},
+		// Valid v0 P2WSH.
+		{
+			sigScript: nil,
+			witness: [][]byte{
+				[]byte{},
+				// Witness script.
+				[]byte{
+					0x21, 0x03, 0x82, 0x62, 0xa6, 0xc6,
+					0xce, 0xc9, 0x3c, 0x2d, 0x3e, 0xcd,
+					0x6c, 0x60, 0x72, 0xef, 0xea, 0x86,
+					0xd0, 0x2f, 0xf8, 0xe3, 0x32, 0x8b,
+					0xbd, 0x02, 0x42, 0xb2, 0x0a, 0xf3,
+					0x42, 0x59, 0x90, 0xac, 0xac,
+				},
+			},
+			class: WitnessV0ScriptHashTy,
+			pkScript: []byte{
+				// OP_0
+				0x00,
+				// OP_DATA_32
+				0x20,
+				// <32-byte script hash>
+				0x01, 0xd5, 0xd9, 0x2e, 0xff, 0xa6, 0xff, 0xba,
+				0x3e, 0xfa, 0x37, 0x9f, 0x98, 0x30, 0xd0, 0xf7,
+				0x56, 0x18, 0xb1, 0x33, 0x93, 0x82, 0x71, 0x52,
+				0xd2, 0x6e, 0x43, 0x09, 0x00, 0x0e, 0x88, 0xb1,
+			},
+		},
+		// Valid v0 P2WPKH.
+		{
+			sigScript: nil,
+			witness: [][]byte{
+				// Signature is not needed to re-derive the
+				// pkScript.
+				[]byte{},
+				// Compressed pubkey.
+				[]byte{
+					0x03, 0x82, 0x62, 0xa6, 0xc6, 0xce,
+					0xc9, 0x3c, 0x2d, 0x3e, 0xcd, 0x6c,
+					0x60, 0x72, 0xef, 0xea, 0x86, 0xd0,
+					0x2f, 0xf8, 0xe3, 0x32, 0x8b, 0xbd,
+					0x02, 0x42, 0xb2, 0x0a, 0xf3, 0x42,
+					0x59, 0x90, 0xac,
+				},
+			},
+			class: WitnessV0PubKeyHashTy,
+			pkScript: []byte{
+				// OP_0
+				0x00,
+				// OP_DATA_20
+				0x14,
+				// <20-byte pubkey hash>
+				0x1d, 0x7c, 0xd6, 0xc7, 0x5c, 0x2e, 0x86, 0xf4,
+				0xcb, 0xf9, 0x8e, 0xae, 0xd2, 0x21, 0xb3, 0x0b,
+				0xd9, 0xa0, 0xb9, 0x28,
+			},
+		},
+		// Invalid v0 P2WPKH - same as above but missing a byte on the
+		// public key.
+		{
+			sigScript: nil,
+			witness: [][]byte{
+				// Signature is not needed to re-derive the
+				// pkScript.
+				[]byte{},
+				// Malformed compressed pubkey.
+				[]byte{
+					0x03, 0x82, 0x62, 0xa6, 0xc6, 0xce,
+					0xc9, 0x3c, 0x2d, 0x3e, 0xcd, 0x6c,
+					0x60, 0x72, 0xef, 0xea, 0x86, 0xd0,
+					0x2f, 0xf8, 0xe3, 0x32, 0x8b, 0xbd,
+					0x02, 0x42, 0xb2, 0x0a, 0xf3, 0x42,
+					0x59, 0x90,
+				},
+			},
+			class:    WitnessV0PubKeyHashTy,
+			pkScript: nil,
+		},
+	}
+
+	for i, test := range tests {
+		valid := test.pkScript != nil
+		pkScript, err := ComputePkScript(test.sigScript, test.witness)
+		if err != nil && valid {
+			t.Fatalf("test #%d: unable to compute pkScript: %v", i, err)
```

Comment:
nit: use `t.Run()` and give each test case a name, that way we avoid having to prefix `"test #%d: "` to every failure message

Reply:
Nice, fixed.